hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamBuilder.java

```diff
@@ -590,7 +589,7 @@ public Output build() {
                     .traceData(TraceData.newBuilder().evmTraceData(builder))
                     .build());
         }
-        if (!stateChanges.isEmpty()) {
+        if (!stateChanges.isEmpty() || topLevel) {
```

Comment:
If this is the place we added one state change for any top level transaction, we should probably add comment :)

Reply:
Yes will do in follow-up PR. 🙏